### PR TITLE
Added Event For Organization Extension Creation

### PR DESCRIPTION
**Pull request by joffinjoy**

### CODE REVIEW

File: src/constants/eventEndpoints.js [hunk 1]
```diff
@@ -35,5 +35,12 @@ module.exports = {
 				route: '/mentoring/v1/sessions/bulkUpdateMentorNames',
 			},
 		],
+		createOrganizationExtension: [
```
Comment (aks30) [inline src/constants/eventEndpoints.js:38]: @joffinjoy did we not confirm that all complete URLs can be given via envs? And multiple URLs for same event can be there?

**Review by aks30 — CHANGES_REQUESTED** [review]
Reviewed on 22 Jan

**Comment by joffinjoy** [comment]
New envs:

EVENT_GROUP_ORGANIZATION='organizationEvents'
EVENTS_ALLOWED_ARGS='updated_at,created_at,update_by,created_by'
EVENT_TYPE_CREATE='update'
EVENT_TYPE_UPDATE='create'
EVENT_TYPE_DELETE='delete'
EVENT_ENTITY_ORGANIZATION='organization'

File: src/.env.sample [hunk 2]
```diff
@@ -182,3 +182,11 @@ SCHEDULER_SERVICE_BASE_URL= '/scheduler/'
 
 #Refresh interval for materialized views
 REFRESH_VIEW_INTERVAL= 540000
+
+
+EVENT_GROUP_ORGANIZATION='organizationEvents'
+EVENTS_ALLOWED_ARGS='updated_at,created_at,update_by,created_by'
```
Comment (joffinjoy) [inline src/.env.sample:188]: Move to hardcode

File: src/.env.sample [hunk 3]
```diff
@@ -182,3 +182,11 @@ SCHEDULER_SERVICE_BASE_URL= '/scheduler/'
...
+EVENT_TYPE_CREATE='create'
+EVENT_TYPE_UPDATE='update'
+EVENT_TYPE_DELETE='delete'
```
Comment (joffinjoy) [inline src/.env.sample:191]: Remove these ENVs

File: src/dtos/eventBody.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+'use strict'
+exports.eventBodyDTO = ({ entity, eventType, entityId, changedValues = [], args = {} }) => {
+	try {
+		if (!entity || !eventType || !entityId)
+			throw new Error('Entity, EventType & EntityId values are mandatory for an Event')
```
Comment (joffinjoy) [inline src/dtos/eventBody.js:5]: This shouldn't throw the error to the caller.

File: src/.env.sample [hunk 5]
```diff
@@ -159,5 +159,8 @@ SCHEDULER_SERVICE_BASE_URL= '/scheduler/'
 #Refresh interval for materialized views
 REFRESH_VIEW_INTERVAL= 540000
 
+
+EVENT_ORG_LISTENER_URLS='http://interface:3567/mentoring/v1/organization/eventListener'
```
Comment (rakeshSgr) [inline src/.env.sample:163]: @joffinjoy can we divide this into two parts, base URL  and endpoint

Comment (joffinjoy) [reply src/.env.sample:163]: Akash wanted the full paths separated by commas.

Comment (rakeshSgr) [reply src/.env.sample:163]: @joffinjoy I'm thinking, endpoints can be in common it won't change most of the time

File: src/generics/form.js [hunk 6]
```diff
@@ -1,11 +1,13 @@
 const formsQueries = require('@database/queries/form')
-const utils = require('@generics/utils')
+//const utils = require('@generics/utils')
```
Comment (rakeshSgr) [inline src/generics/form.js:2]: @joffinjoy can you remove unwanted code

Comment (joffinjoy) [reply src/generics/form.js:2]: Can we keep it for now. The thing is, we need this code to use the caching itself. But in current form, introducing caching will introduce circular dependency. I will look into this issue as a separate one. Just want to retain this form for that.

**Pull request merged**